doc/specs/vulkan/appendices/debug_marker.txt

```diff
@@ -0,0 +1,163 @@
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_POOL_EXT is a sname:VkDescriptorPool.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_SET_EXT is a sname:VkDescriptorSet.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_FRAMEBUFFER_EXT is a sname:VkFramebuffer.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_POOL_EXT is a sname:VkCommandPool.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_SURFACE_KHR_EXT is a sname:VkSurfaceKHR.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_SWAPCHAIN_KHR_EXT is a sname:VkSwapchainKHR.
+  * ename:VK_DEBUG_REPORT_OBJECT_TYPE_DEBUG_REPORT_EXT is a sname:VkDebugReportCallbackEXT.
+
+
+== Command Buffer Markers
+
+Typical Vulkan applications will submit many command buffers in each frame,
+with each command buffer containing a large number of individual commands.
+Being able to logically annotate regions of command buffers that belong
+together as well as hierarchically subdivide the frame is important to
+a developer's ability to navigate the commands viewed holistically.
+
+The marker commands fname:vkCmdDebugMarkerBeginEXT and
+fname:vkCmdDebugMarkerEndEXT define regions of a series of commands that are
+grouped together, and they can be nested to create a hierarchy. The
+fname:vkCmdDebugMarkerInsertEXT command allows insertion of a single label
+within a command buffer.
+
+A marker region can be opened by calling:
+
+include::../protos/vkCmdDebugMarkerBeginEXT.txt[]
+
+  * pname:commandBuffer is the command buffer into which the command is
+    recorded.
+  * pname:pMarker is a pointer to a null-terminated UTF-8 string that
+    contains the name of the region to open.
+
+include::../validity/protos/vkCmdDebugMarkerBeginEXT.txt[]
+
+A marker region can be closed by calling:
+
+include::../protos/vkCmdDebugMarkerEndEXT.txt[]
+
+  * pname:commandBuffer is the command buffer into which the command is
+    recorded.
+
+include::../validity/protos/vkCmdDebugMarkerEndEXT.txt[]
+
+An application may: open a marker region in one command buffer and close it
+in another, or otherwise split marker regions across multiple command
+buffers or multiple queue submissions. When viewed from the linear series of
+submissions to a single queue, the calls to fname:vkCmdDebugMarkerBeginEXT
+and fname:vkCmdDebugMarkerEndEXT must: be matched and balanced.
+
+Any calls to fname:vkCmdDebugMarkerBeginEXT within a secondary command
+buffer must have a matching fname:vkCmdDebugMarkerEndEXT in that same
+command buffer, and marker regions begun outside of the secondary command
+buffer must not be ended inside it.
+
+A single marker label can be inserted into a command buffer by calling:
+
+include::../protos/vkCmdDebugMarkerInsertEXT.txt[]
+
+  * pname:commandBuffer is the command buffer into which the command is
+    recorded.
+  * pname:pMarker is a pointer to a null-terminated UTF-8 string that
+    contains the name of the marker to insert.
+
+include::../validity/protos/vkCmdDebugMarkerInsertEXT.txt[]
```

Comment:
Where / how does a developer see these markers? Is that something that API dump needs to print?
Are the markers somehow placed in the actual command buffers (would seem not if this is just a validation layer feature).